packages/backend/src/workers/action.ts

```diff
@@ -35,6 +43,18 @@ export const worker = new Worker(
     const jobOptions = {
       removeOnComplete: REMOVE_AFTER_7_DAYS_OR_50_JOBS,
       removeOnFail: REMOVE_AFTER_30_DAYS_OR_150_JOBS,
+      delay: DEFAULT_DELAY_DURATION,
+    };
+
+    const stepApp = await step.getApp();
+
+    if (stepApp.key === 'delay') {
```

Comment:
```suggestion
    if (step.appKey === 'delay') {
```

@farukaydin, what do you think about leveraging the step itself, rather than its app, and removing line 49?